californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java

```diff
@@ -233,7 +233,12 @@ public void receiveRequest(final Request request, final EndpointReceiver receive
 
 				@Override
 				public void run() {
-					receiver.receiveRequest(previous, request);
+					try {
+						receiver.receiveRequest(previous, request);
+					} catch (Exception ex) {
+						LOGGER.debug("error receiving request {} again!", request, ex);
+						receiver.reject(request);
```

Comment:
I'm not sure if sending RST is a good idea... :thinking: 

Did you consider to send an Internal Server Error Response ?